src/storage/txn/commands/prewrite.rs

```diff
@@ -587,27 +592,32 @@ impl PrewriteKind for Optimistic {
     }
 }
 
+/// Information needed for the pessimistic variant of `PreWriteKind`.
 struct Pessimistic {
     for_update_ts: TimeStamp,
 }
 
+/// The pessimistic variant of `PreWriteKind`.
 impl PrewriteKind for Pessimistic {
+    // see the `MutationLock` trait implementation for `(Mutation, bool)`
```

Comment:
ditto

Reply:
I agree. I think after documenting both the Trait and the Struct, we can skip the doc for the impl.